### PR TITLE
DOCS-#3873: fix 'more than one target found for cross-reference DataFrame' warning

### LINKED ISSUE #3873: DOCS: WARNING more than one target found for cross-reference 'DataFrame': modin.experimental.pandas.DataFrame, modin.pandas.dataframe.DataFrame

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3873 
- [x] tests ~added~ and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/flow/modin/experimental/pandas.rst [hunk 1]
```diff
@@ -12,5 +12,4 @@ Experimental API Reference
 .. autofunction:: read_sql
 .. autofunction:: read_csv_glob
 .. autofunction:: read_pickle_distributed
-.. autoclass:: DataFrame
-   :members: to_pickle_distributed
```
Comment (anmyachev) [inline docs/flow/modin/experimental/pandas.rst:16]: There is no need to show `DataFrame` class documentation.

Comment (anmyachev) [reply docs/flow/modin/experimental/pandas.rst:16]: The new implementation does not create another target with the `DataFrame` name.

**Review by dchigarev — APPROVED** [review]
LGTM! The warnings have disappeared indeed ([build log](https://readthedocs.org/projects/modin/builds/15585834/)).

**Review by devin-petersohn — APPROVED** [review]
LGTM!

**Pull request merged**